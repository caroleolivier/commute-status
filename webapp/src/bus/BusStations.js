import React, { Component } from 'react';
import PropTypes from 'prop-types';

import BusStationContainer from './BusStationContainer';

class BusStations extends Component {
    render() {
        const busStations = [];
        for (let i = 0; i < this.props.busStations.length; i += 1) {
            const busStation = this.props.busStations[i];
            const key = busStation.stationId;
            busStations.push(
                <BusStationContainer key={key} busStation={busStation} />
            );
        }
        return (
            <div>
                <h3>Buse(s)</h3>
                {busStations}
            </div>
        );
    }
}

BusStations.propTypes = {
    busStations: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default BusStations;
