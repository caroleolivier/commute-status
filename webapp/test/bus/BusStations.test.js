/* eslint-env browser, jest */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BusStations from '../../src/bus/BusStations';

describe('BusStations component', () => {
    test('shallow render matches snapshot.', () => {
        const busStations = [
            { stationName: 'Houston', stationId: '12345', direction: 'Outter Space', buses: ['1', '2'] },
            { stationName: 'Moon', stationId: '8372hf8', direction: 'Saturne', buses: ['N12'] }
        ];
        const busStationContainer = shallow(<BusStations busStations={busStations} />);
        const tree = toJson(busStationContainer);
        expect(tree).toMatchSnapshot('rendering');
    });
});
