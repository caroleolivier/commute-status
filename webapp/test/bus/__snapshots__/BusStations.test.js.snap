// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<div>
  <h3>
    Direction: 
    Neptune
  </h3>
  <BusStationContainer
    busStation={
      Object {
        "buses": Array [
          "1",
          "2",
        ],
        "direction": "Outter Space",
        "stationId": "12345",
        "stationName": "Houston",
      }
    }
    key="12345"
  />
  <BusStationContainer
    busStation={
      Object {
        "buses": Array [
          "N12",
        ],
        "direction": "Saturne",
        "stationId": "8372hf8",
        "stationName": "Moon",
      }
    }
    key="8372hf8"
  />
</div>
`;
